packages/mui-base/src/useAutocomplete/useAutocomplete.js

```diff
@@ -236,7 +236,15 @@ export default function useAutocomplete(props) {
   });
 
   React.useEffect(() => {
-    const valueChange = value !== previousProps.value;
+    // The value can be an instance of object and thus comparing it with basic equality lead to
+    // invalid results. Instead we should compare its final label to detect a value change.
+    const valueChange =
+      value !== previousProps.value &&
+      (previousProps.value === undefined ||
+        previousProps.value === null ||
+        value === undefined ||
+        value === null ||
+        !isOptionEqualToValue(value, previousProps.value));
```

Comment:
`isOptionEqualToValue` is meant to compare the selected value with options, not options themselves. The problem you're describing here could be solved by the introduction of another prop (something like `areOptionsEqual`). However, to justify introduction of a new prop, we need to gather interest/pain in the community. I suggest to close this PR and create an issue that describes the problem, discuss the possible solutions and wait for upvotes from the community.

Reply:
Definitely will do! Thanks for the suggestion.